### PR TITLE
rename seismic.offset to seismic.stacking_offset

### PR DESCRIPTION
**Pull request by perolavsvendsen**
Subtask of #212

### CODE REVIEW

**Review by ashildskalnes — COMMENTED** [review]
Looks OK to me:-)

**Review by ezaezaeza — COMMENTED** [review]
stacking offset is better than offset. However, to me it still seems a bit odd to add stacking offset as metadata to the surface, rather than linking it to the source of the attribute map.i.e. the seismic cube. This is a similar topic for a time surface, I guess , that the metadata say something about which cube (incl stacking offset) was used in the interpretation.  Also the attribute of the seismic cube (amplitude, relai, timeshift..) should be an attribute connected to the source (seismic cube). To have this info in the meta data for a seismic map is maybe more important  than the stacking offset. As I understand this infomation (amplitude, relai..) is only specified in the tag name?

**Comment by perolavsvendsen** [comment]
> stacking offset is better than offset. However, to... 

Yes, there is a bigger discussion here - but that should probably be taken in the main issue (#212). This PR is only renaming seismic.offset to seismic.stacking_offset, ref action from our discussion the other day.

**Review by jcrivenaes — APPROVED** [review]
LGTM

**Comment by perolavsvendsen** [comment]
Thinking about keeping the old as an alias with a depreciation warning in case, in case this is actually being used.

**Comment by ashildskalnes** [comment]
I estimate that the probability for this being already used is 0.0001 % so I don't hink that you need to bother with keeping it as an alias:-)

**Comment by perolavsvendsen** [comment]
> Ok, but what I actually meant was that this is the probability for a **client app using this metadata** for something

Ah, OK. Yes, agree, that is extremely unlikely. But the issue I suddenly got a bit worried about, and possibly have confirmed now, is that `seismic.offset` is given as input argument in existing scripts. When we rename it to `seismic.stacking_offset` dataio will crash due to illegal argument.

**Review by ezaezaeza — APPROVED** [review]
LGTM (prøver meg med ny sjargong:-))

File: tests/test_units/test_dataio.py [hunk 1]
```diff
@@ -144,6 +154,42 @@ def test_content_is_a_wrongly_formatted_dict(globalconfig2):
...
+            content={
+                "seismic": {
+                    "offset": "0-15",
+                }
+            },
+            verbosity="DEBUG",
+        )
+        mymeta = eobj.generate_metadata(regsurf)
+
+    # depreacted 'offset' replaced with 'stacking_offset'
```
Comment (jcrivenaes) [inline tests/test_units/test_dataio.py:186]: typo -> deprecated

File: src/fmu/dataio/dataio.py [hunk 2]
```diff
@@ -192,9 +199,29 @@ def _content_validate(name, fields):
...
+
+            message = f"Content {name}.{key} is deprecated. "
+
+            if replaced_by is not None:
+                message += f"Please use {replaced_by}. "
+                replace_deprecated.update({key: replaced_by})
+
+            warn(
+                message,
+                PendingDeprecationWarning,
```
Comment (jcrivenaes) [inline src/fmu/dataio/dataio.py:214]: Better with just `DeprecationWarning`? Do not see the ned for "pending" here.

**Review by jcrivenaes — APPROVED** [review]
Consider comments, othewise seems OK

File: src/fmu/dataio/dataio.py (hunk 2 above)
Comment (perolavsvendsen) [reply src/fmu/dataio/dataio.py:214]: Copy/paste... I agree, will fix 👍🏻

**Pull request merged**